illumio_connector.py

```diff
@@ -0,0 +1,908 @@
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        try:
+            hrefs_list = self.handle_comma_seperated_string(hrefs)
+
+            provisioned_virtual_service_obj = self._pce.provision_policy_changes(
+                change_description="Object provisioning.",
+                hrefs=hrefs_list,
+            )
+        except Exception as e:
+            self.error_print("Failed to provision objects: {}".format(str(e)))
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                "Encountered error provisioning object: {}".format(e),
+            )
+
+        result = self.convert_object_to_json(
+            provisioned_virtual_service_obj, action_result
+        )
+        provisioned_href_list = [
+            illumio.convert_draft_href_to_active(href) for href in hrefs_list
+        ]
+        result["provisioned_href"] = provisioned_href_list
+        action_result.add_data(result)
+        return action_result.set_status(
+            phantom.APP_SUCCESS, "Successfully provisioned object"
+        )
+
+    def _handle_get_ip_lists(self, param):
+        name = param.get("name")
+        description = param.get("description")
+        fqdn = param.get("fqdn")
+        ip_address = param.get("ip_address")
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        ret_val = self.connect_pce(action_result)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        try:
+            ip_list = self._pce.ip_lists.get(
+                params={
+                    "name": name,
+                    "description": description,
+                    "fqdn": fqdn,
+                    "ip_address": ip_address,
+                }
+            )
+            self.debug_print("Found {} IP lists".format(len(ip_list)))
+            output_msg = "Successfully fetched IP List" if ip_list else "No Data Found"
+            action_result.set_status(phantom.APP_SUCCESS, output_msg)
+        except Exception as e:
+            self.error_print("Failed to get IP lists: {}".format(str(e)))
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                "Encountered error fetching IP List: {}".format(e),
+            )
+
+        result = {
+            "ip_lists": [
+                self.convert_object_to_json(lists, action_result) for lists in ip_list
+            ]
+        }
+        action_result.add_data(result)
+        return action_result.get_status()
+
+    def _handle_create_ruleset(self, param):
+        name = param["name"]
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        rule_set = None
+
+        ret_val = self.connect_pce(action_result)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        try:
+            self.debug_print("Creating rule set '{}'".format(name))
```

Comment:
Rule set is two words here, but in the JSON, it is referred to as ruleset. Please keep it consistent